src/c++/perf_analyzer/test_request_rate_manager.cc

```diff
@@ -53,10 +54,36 @@ class TestRequestRateManager : public TestLoadManagerBase,
             params.batch_size, params.measurement_window_ms, params.max_threads,
             params.num_of_sequences, params.sequence_length,
             params.shared_memory_type, params.output_shm_size,
-            params.start_sequence_id, params.sequence_id_range,
-            params.string_length, params.string_data, params.zero_input,
-            params.user_data, GetParser(), GetFactory())
+            params.start_sequence_id, params.sequence_id_range, GetParser(),
+            GetFactory())
   {
+    InitManager(
+        params.string_length, params.string_data, params.zero_input,
+        params.user_data);
+  }
+
+  TestRequestRateManager(
+      PerfAnalyzerParameters params, const std::shared_ptr<ModelParser>& parser,
+      bool is_sequence_model = false, bool is_decoupled_model = false,
+      bool use_mock_infer = false)
+      : use_mock_infer_(use_mock_infer),
+        TestLoadManagerBase(params, is_sequence_model, is_decoupled_model),
+        RequestRateManager(
+            params.async, params.streaming, params.request_distribution,
+            params.batch_size, params.measurement_window_ms, params.max_threads,
+            params.num_of_sequences, params.sequence_length,
+            params.shared_memory_type, params.output_shm_size,
+            params.start_sequence_id, params.sequence_id_range, parser,
+            GetFactory())
+  {
```

Comment:
this one doesn't call initmanager?
Just to confirm, the only difference is a parser passed in?